### ISSUE TITLE
add_bucket_to_path may construct path incorrectly on windows machines

### ISSUE THREAD

**Issue by johns10** [ex-aws/ex_aws#746]
### Environment
IEx 1.11.2 (compiled with Erlang/OTP 21) (second computer I've tried this on)
This only happens in Windows, possibly because of how paths are constructed.  I tried the exact same operations on my wsl instance, and it ran fine.  I've tried this on two windows machines.
{:ex_aws, "> 2.1.7"},
{:hackney, "> 1.9"},`

using with
{:waffle, "> 1.1.3"},
{:waffle_ecto, "> 0.0.9"},
{:ex_aws_s3, "> 2.1.0"},
{:hackney, "> 1.9"},
{:sweet_xml, "> 0.6"

### Current behavior
When using this library, I can't put S3 files, or interact through the S3 API at all.  
When I run
`ExAws.S3.list_buckets() |> ExAws.request!()`
I get 
```
(ArgumentError) :path in URI must be nil or an absolute path if :host or :authority are given, got: %URI{authority: nil, fragment: nil, host: "s3.us-east-2.amazonaws.com", path: "c:/", port: 443, query: "", scheme: "https", userinfo: nil}
    (elixir 1.11.2) lib/uri.ex:662: String.Chars.URI.to_string/1
    (ex_aws 2.1.7) lib/ex_aws/request/url.ex:19: ExAws.Request.Url.build/2
    (ex_aws 2.1.7) lib/ex_aws/operation/s3.ex:36: ExAws.Operation.ExAws.Operation.S3.perform/2
    (ex_aws 2.1.7) lib/ex_aws.ex:61: ExAws.request!/2
```
Similarly, when I try to put files using the Waffle Ecto API, like this: 

`UserDocs.Media.update_screenshot(screenshot, %{ aws_file: (@dev_path <> filename)})` 

with these options on the changeset: 

`cast_attachments(attrs, [:aws_file], allow_paths: true)`
I get
```** (ArgumentError) :path in URI must be nil or an absolute path if :host or :authority are given, got: %URI{authority: nil, fragment: nil, host: "s3.us-east-2.amazonaws.com", path: "c:/userdocs-screenshots/uploads/Setup Manager Session 35.jpeg", port: 443, query: "uploads=1", scheme: "https", userinfo: nil}```
I think the problem is the :path arg is getting a full path, when the call to URL.build expects a relative path.  It's expecting `/uploads/Setup Manager Session 35.jpeg`

I traced this behavior back to this function:
```
def add_bucket_to_path(operation, config) do
      path = Path.join(["/", operation.bucket, operation.path]) |> Path.expand()
      {operation |> Map.put(:path, path), config}
    end
```
It works on my local by removing the call to Path.expand().  I'm not sure if this is applicable to other platforms though.  This may only be happening because of the way the path is constructed on PC's vs UNIX systems.
```
def add_bucket_to_path(operation, config) do
      path = Path.join(["/", operation.bucket, operation.path])
      {operation |> Map.put(:path, path), config}
    end
```

**Comment by bernardd** [comment]
Good catch - thanks for taking the time to pin down the issue. I've pushed a fix in #749 which keeps the resolution behaviour (expansion of `.` and `..` paths) but should work safely on Windows. Please let me know how you get on with it.

**Comment by losvedir** [comment]
We just ran into this issue in our Windows app as well. The current master branch (this [commit](https://github.com/ex-aws/ex_aws/commit/4d161180f5447ccb872c81527a45ac8a98c51820)) has resolved the issue for us.
